apps/files/src/mixins/collaborators.js

```diff
@@ -78,6 +98,16 @@ export default {
         .finally(_ => {
           this.loading = false
         })
+    },
+
+    collaboratorOptionChanged ({ role, permissions, propagate = true }) {
```

Comment:
I suggest to have an attribute on the collaboratorRoles file that tells the name of the default role.

Then the parent view can pre-set the default role to `this.selectedRole` to avoid the "propagate = false" gymnastics.

Reply:
also I assume that the roles API in the future will also provide a field for the default role (or we make it default to the one with the least permissions?)

Reply:
=> later